lib/factories.js

```diff
@@ -0,0 +1,125 @@
+    overrides.isDisplayed !== undefined ? overrides.isDisplayed : true,
+    overrides.isSendingAlerts !== undefined ? overrides.isSendingAlerts : true,
+    overrides.isSendingVitals !== undefined ? overrides.isSendingVitals : true,
+    overrides.language !== undefined ? overrides.language : 'en',
+    overrides.createdAt !== undefined ? overrides.createdAt : new Date(),
+    overrides.updatedAt !== undefined ? overrides.updatedAt : new Date(),
+  )
+}
+
+async function deviceDBFactory(db, overrides = {}) {
+  const device = await db.createDevice(
+    overrides.id !== undefined ? overrides.id : 'fakeDeviceId',
+    overrides.deviceType !== undefined ? overrides.deviceType : DEVICE_TYPE.DEVICE_BUTTON,
+    overrides.locationid !== undefined ? overrides.locationid : null,
+    overrides.phoneNumber !== undefined ? overrides.phoneNumber : '+12223334444',
+    overrides.displayName !== undefined ? overrides.displayName : 'Unit 305',
+    overrides.serialNumber !== undefined ? overrides.serialNumber : 'AB12-12345',
+    overrides.sentLowBatteryAlertAt !== undefined ? overrides.sentLowBatteryAlertAt : null,
+    overrides.sentVitalsAlertAt !== undefined ? overrides.sentVitalsAlertAt : null,
+    overrides.createdAt !== undefined ? overrides.createdAt : new Date(),
+    overrides.updatedAt !== undefined ? overrides.updatedAt : new Date(),
+    overrides.isDisplayed !== undefined ? overrides.isDisplayed : true,
+    overrides.isSendingAlerts !== undefined ? overrides.isSendingAlerts : true,
+    overrides.isSendingVitals !== undefined ? overrides.isSendingVitals : true,
+    overrides.clientId !== undefined ? overrides.clientId : 'fakeClientId',
+  )
+
+  return device
+}
+
+function deviceFactory(db, overrides = {}) {
+  return new Device(
+    overrides.id !== undefined ? overrides.id : 'fakeDeviceId',
+    overrides.deviceType !== undefined ? overrides.deviceType : DEVICE_TYPE.DEVICE_BUTTON,
+    overrides.locationid !== undefined ? overrides.locationid : null,
+    overrides.phoneNumber !== undefined ? overrides.phoneNumber : '+12223334444',
+    overrides.displayName !== undefined ? overrides.displayName : 'Unit 305',
+    overrides.serialNumber !== undefined ? overrides.serialNumber : 'AB12-12345',
+    overrides.sentLowBatteryAlertAt !== undefined ? overrides.sentLowBatteryAlertAt : null,
+    overrides.sentVitalsAlertAt !== undefined ? overrides.sentVitalsAlertAt : null,
+    overrides.createdAt !== undefined ? overrides.createdAt : new Date(),
+    overrides.updatedAt !== undefined ? overrides.updatedAt : new Date(),
+    overrides.isDisplayed !== undefined ? overrides.isDisplayed : true,
+    overrides.isSendingAlerts !== undefined ? overrides.isSendingAlerts : true,
+    overrides.isSendingVitals !== undefined ? overrides.isSendingVitals : true,
+    overrides.client !== undefined ? overrides.client : clientFactory(),
```

Comment:
Ah, looks like you were sending an object to the constructor already. Just make sure that this order matches what you have in `Device.js` :smile: